### DISCUSSION TITLE
Get Redis connection in Decorator

### DISCUSSION THREAD

**Discussion by joeynq** [TypedProject/ts-express-decorators#2471] [Q&A] [answered]
Hi @Romakita,

I need to access to `REDIS_CONNECTION` on my custom decorators, as the codes below.

```ts
export function MyDecorator(): ClassDecorator {
  return useDecorators(
    JsonEntityFn((store: JsonEntityStore, args: DecoratorParameters) => {
      // I need to get REDIS_CONNECTION from here, connection already set in Configuration
    }
  )
}

// usage
@MyDecorator() // this read data from redis to perform some operation to the controller
class MyController {}
```

Please guide me through how to achieve that.

Many thanks,

**Comment by Romakita** [comment]
Hello @datnq 
You misunderstood the princible of decorator. His goal is to separate metadata from the technical code and the business code.

the normal flow is to set a metada related to a class and create a module or service to get the class & his metadata to implement something.

also the connection is outside of the decorator scope. The DI container is also outside of this scope. 

So I won’t be able to help you and I won’t encourage you in this way. There is no reason to do that directly in a decorator! ( design pattern etc.)

why do you need that? Maybe I can help you to find the corect way to do that!

see you
Romain

**Reply by joeynq** [reply]
I see what you mean and you're correct.

My purpose is collect settings of some other decorators and save it to Redis and process those data by another services. I want those are stored as soon as the server started.

Now I'm thinking of two ways.

I think what I need is save those to some Store and get them when all controllers initiated, `$afterRoutesInit` I think?

Or, I can get Json metadata of all controllers `$afterRoutesInit`

What is the proper way to do that? Please share your advice.

Thanks,

**Reply by joeynq** [reply]
@Romakita I changed the code as you suggested and get expected result just by using `StoreSet`
Thank you very much

**Reply by Romakita** [reply, marked as answer]
Hello @datnq 
You can take this PR as example:
https://github.com/tsedio/tsed/pull/2464/files
 It uses custom decorator and consume metadata inside an injectable service ;)

See you
